internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -174,6 +188,81 @@ func (c *cmd) GetCommandName() string {
 	return CommandName
 }
 
+// setupAgentToken is to set up the agent token to the running agent if haven't set up yet
```

Comment:
Comment that side effect is `BootStrapACLTokenInfo` is set if passed in as nil

Reply:
Actual, this doesn't work. 
The pointer itself is passed by value, so changing what it points to in the function does change the callers copy of the pointer. Best approach is to add `BootStrapACLTokenInfo` as a return value. Then the caller can use the returned value .

Reply:
i am not changing what it points in this func. I have a new variable defined.

Reply:
Right, just want to make sure. Does the caller need to new `BootStrapACLTokenInfo ` in this case?

Reply:
no need, if the inputToken is nil (not new'ed), then it reconstructs it from the saved file- i've added the comments in this func.